pkgs/top-level/by-name-overlay.nix

```diff
@@ -32,27 +36,35 @@ let
       # Additionally in either of those alternatives, we would have to duplicate the hardcoding of "README.md"
       { }
     else
-      mapAttrs (name: _: baseDirectory + "/${shard}/${name}/package.nix") (
+      mapAttrs (name: _: import (baseDirectory + "/${shard}/${name}/package.nix")) (
         readDir (baseDirectory + "/${shard}")
       );
 
-  # The attribute set mapping names to the package files defining them
+  # The attribute set mapping names to the package expressions defining them
   # This is defined up here in order to allow reuse of the value (it's kind of expensive to compute)
   # if the overlay has to be applied multiple times
-  packageFiles = mergeAttrsList (mapAttrsToList namesForShard (readDir baseDirectory));
+  packages = mergeAttrsList (mapAttrsToList namesForShard (readDir baseDirectory));
 in
 # TODO: Consider optimising this using `builtins.deepSeq packageFiles`,
 # which could free up the above thunks and reduce GC times.
 # Currently this would be hard to measure until we have more packages
 # and ideally https://github.com/NixOS/nix/pull/8895
 self: super:
 {
+  _isAlias = name: packages.${name}.type or null == "alias";
```

Comment:
I think we should use double underscores for this, to match `__spliced` and the like.